src/Symfony/Component/Scheduler/Generator/MessageGenerator.php

```diff
@@ -24,13 +24,16 @@ final class MessageGenerator implements MessageGeneratorInterface
     private TriggerHeap $triggerHeap;
     private ?\DateTimeImmutable $waitUntil;
     private CheckpointInterface $checkpoint;
+    private ?string $name;
 
     public function __construct(
         private readonly Schedule $schedule,
         string|CheckpointInterface $checkpoint,
         private readonly ClockInterface $clock = new Clock(),
+        string $name = null,
     ) {
         $this->waitUntil = new \DateTimeImmutable('@0');
+        $this->name = $name ?? (\is_string($checkpoint) ? $checkpoint : null);
         if (\is_string($checkpoint)) {
             $checkpoint = new Checkpoint('scheduler_checkpoint_'.$checkpoint, $this->schedule->getLock(), $this->schedule->getState());
         }
```

Comment:
I'm wondering if it would make more sense to require a name and make the checkpoint optional?
If this is something we want, as the component is not yet released and is experimental we can change the signature easily. For example:

```suggestion
    public function __construct(
        private readonly Schedule $schedule,
        private readonly string $name,
        private readonly ClockInterface $clock = new Clock(),
        ?CheckpointInterface $checkpoint = null,
    )
    {
        $this->waitUntil = new \DateTimeImmutable('@0');
        $this->checkpoint = $checkpoint ?? new Checkpoint('scheduler_checkpoint_' . $this->name, $this->schedule->getLock(), $this->schedule->getState());
    }
```

Reply:
(If so, it means the name will always be here and so we can require it for `MessageContext`)

Reply:
I like this idea!